### PR TITLE
Shorten Envoy Mobile error details

### PR DESCRIPTION
**Pull request by RenjieTang**
Commit Message: Shorten Envoy Mobile error details
Additional Description: This PR also changes , to | for one of the error string.
Risk Level: Low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: mobile/library/common/http/client.cc [hunk 1]
```diff
@@ -442,34 +442,34 @@ void Client::DirectStreamCallbacks::latchError() {
   if (info.responseCode().has_value()) {
     error_->error_code_ = Bridge::Utility::errorCodeFromLocalStatus(
         static_cast<Http::Code>(info.responseCode().value()));
-    error_msg_details.push_back(absl::StrCat("RESPONSE_CODE: ", info.responseCode().value()));
+    error_msg_details.push_back(absl::StrCat("envoy_rc: ", info.responseCode().value()));
```
Comment (abeyad) [inline mobile/library/common/http/client.cc:445]: `envoy_` seems redundant, just `rc` is fine?

File: mobile/library/common/http/client.cc [hunk 2]
```diff
@@ -442,34 +442,34 @@ void Client::DirectStreamCallbacks::latchError() {
...
   } else if (StreamInfo::isStreamIdleTimeout(info)) {
     error_->error_code_ = ENVOY_REQUEST_TIMEOUT;
   } else {
     error_->error_code_ = ENVOY_STREAM_RESET;
   }
 
-  error_msg_details.push_back(absl::StrCat("ERROR_CODE: ", error_->error_code_));
+  error_msg_details.push_back(absl::StrCat("envoy_ec: ", error_->error_code_));
```
Comment (abeyad) [inline mobile/library/common/http/client.cc:452]: same here, `ec` maybe to make it shorter?

File: mobile/library/common/http/client.cc [hunk 3]
```diff
@@ -442,34 +442,34 @@ void Client::DirectStreamCallbacks::latchError() {
...
   std::vector<std::string> response_flags(info.responseFlags().size());
   std::transform(info.responseFlags().begin(), info.responseFlags().end(), response_flags.begin(),
                  [](StreamInfo::ResponseFlag flag) { return std::to_string(flag.value()); });
-  error_msg_details.push_back(absl::StrCat("RESPONSE_FLAGS: ", absl::StrJoin(response_flags, ",")));
+  error_msg_details.push_back(absl::StrCat("rsp_flags: ", absl::StrJoin(response_flags, ",")));
   if (info.protocol().has_value()) {
     // https://github.com/envoyproxy/envoy/blob/fbce85914421145b5ae3210c9313eced63e535b0/envoy/http/protocol.h#L13
-    error_msg_details.push_back(absl::StrCat("PROTOCOL: ", *info.protocol()));
+    error_msg_details.push_back(absl::StrCat("protocol: ", *info.protocol()));
```
Comment (abeyad) [inline mobile/library/common/http/client.cc:459]: we could make this `http`, specifying which http version we're using, saves 4 chars, but `protocol` is a bit clearer.

File: mobile/library/common/http/client.cc [hunk 4]
```diff
@@ -442,34 +442,34 @@ void Client::DirectStreamCallbacks::latchError() {
   if (info.responseCode().has_value()) {
     error_->error_code_ = Bridge::Utility::errorCodeFromLocalStatus(
         static_cast<Http::Code>(info.responseCode().value()));
-    error_msg_details.push_back(absl::StrCat("RESPONSE_CODE: ", info.responseCode().value()));
```
Comment (abeyad) [inline mobile/library/common/http/client.cc:445]: @RenjieTang Are we dealing with details string size limitations that we want to shorten the prefixes?

Comment (RenjieTang) [reply mobile/library/common/http/client.cc:445]: There's a general size limit on the ping data we send to the servers. Strings from all other sources are shortened so we should perform this optimization too.

**Review by abeyad — APPROVED** [review]

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**